# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/DocAccessible.cpp
# Commit: 6a2ad49b2630
# Full Hash: 6a2ad49b2630c185de3c7571db79961ea6b331bd
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-05-03 09:42:08
# Regressor Bug: 1739560
# File Overlap Count: 1
# Description:
#   Bug 1767169: When processing queued cache updates, ensure Accessibles haven't been detached from the document. r=eeejay
#   
#   In bug 1739560, I moved the processing of queued cache updates before the firing of mutation events.
#   That means that Accessibles removed from the tree might not be shut down (and thus defunct) yet, since we have this limbo "not in document" state.
#   Therefore, as well as skipping defunct Accessibles, we must also skip Accessibles that are no longer in the document.
# ==============================================================================

diff -r 3a04fc935eb5 -r 6a2ad49b2630 accessible/generic/DocAccessible.cpp
--- a/accessible/generic/DocAccessible.cpp	Mon May 02 22:37:24 2022 +0000
+++ b/accessible/generic/DocAccessible.cpp	Mon May 02 22:38:07 2022 +0000
@@ -1424,7 +1424,7 @@
   for (auto iter = mQueuedCacheUpdates.Iter(); !iter.Done(); iter.Next()) {
     LocalAccessible* acc = iter.Key();
     uint64_t domain = iter.UserData();
-    if (!acc->IsDefunct()) {
+    if (acc->IsInDocument() && !acc->IsDefunct()) {
       RefPtr<AccAttributes> fields =
           acc->BundleFieldsForCache(domain, CacheUpdateType::Update);
 
